app/frontend/src/utils/transformUtils.js

```diff
@@ -75,167 +75,96 @@ export function attachAttributesToLinks(formSchemaComponents) {
 }
 
 /**
- * @function getAvailableDates
- * Get All possible dates in given period with Term and Interval
+ * @function getSubmissionPeriodDates
+ * Gets all possible dates for a submission period
  *
- * @param {Integer} keepAliveFor An integer for number of days, that tells form to be open for a particular period
- * @param {String} keepAliveForInterval A string of days,Weeks,months, that tells form to be open for a particular period
- * @param {Object[]} formStartDate An object of Moment JS date, that tells form to be open on a particular day/date
- * @param {Integer} term An integer of number of Days/Weeks OR Years, that tells form to be repeated on a particular period
- * @param {String} interval A string of days,Weeks,months, that tells form to be repeated on a particular period
- * @param {Integer} allowLateTerm An integer of number of Days/Weeks OR Years, that tells form to be allowed for late submission for a particular period
- * @param {String} allowLateInterval A string of days,Weeks,months, that tells form to be allowed for late submission for a particular period
- * @param {Object[]} repeatUntil An object of Moment JS date, that tells form to be finally close after repetition on particular day date
- * @param {String} scheduleType A string one of Manual, ClosingDate OR Period
- * @param {Object[]} closeDate An object of Moment JS date, Forms closing date
- * @returns {Object[]} An object array of Available dates in given period
+ * @param {Integer} keepOpenForTerm A submission period's number of period intervals
+ * @param {String} keepOpenForInterval A submission period's intervals which can be days, weeks, months, or years
+ * @param {Object} openSubmissionDateTime A moment object of the day the form will be open for submissions
+ * @param {Integer} repeatSubmissionTerm A submission period's number of repeat intervals
+ * @param {String} repeatSubmissionInterval A submission period's repeat intervals which can be days, weeks, months, or years
+ * @param {Integer} allowLateTerm A late submission's number of period intervals
+ * @param {String} allowLateInterval A late submission's intervals which can be days, weeks, months, or years
+ * @param {Object} repeatSubmissionUntil A moment of the day that a submission period will stop repeat intervals
+ * @returns {Object} An object array of available dates in given period
  */
-export function getAvailableDates(
-  keepAliveFor = 0,
-  keepAliveForInterval = 'days',
-  formStartDate,
-  term = null,
-  interval = null,
+export function getSubmissionPeriodDates(
+  keepOpenForTerm = 0,
```

Comment:
Renamed these variables to match what is actually found in the form.schedule object